configure.ac

```diff
@@ -148,6 +148,12 @@ if $sse41; then
   AC_DEFINE([HAVE_SSE4_1], [1], [Enable SSE 4.1 instructions])
 fi
 
+AX_CHECK_COMPILE_FLAG([-mfpu=neon], [neon=true], [neon=false], [$WERROR])
+AM_CONDITIONAL([HAVE_NEON], $neon)
+if $neon; then
+  AC_DEFINE([HAVE_NEON], [1], [Enable NEON instructions])
```

Comment:
Surprisingly this breaks OSS-Fuzz: https://oss-fuzz-build-logs.storage.googleapis.com/log-2f0c6bde-a12f-49cc-964d-fed11818127e.txt. It uses a build with a clang++ compiler which obviously accepts `-mfpu=neon` on Intel, and the build files because it does not find a header file:

    Step #3: checking whether C++ compiler accepts -mavx... yes
    Step #3: checking whether C++ compiler accepts -mavx2... yes
    Step #3: checking whether C++ compiler accepts -mfma... yes
    Step #3: checking whether C++ compiler accepts -msse4.1... yes
    Step #3: checking whether C++ compiler accepts -mfpu=neon... yes
    [...]
    Step #3: src/arch/simddetect.cpp:46:10: fatal error: 'asm/hwcap.h' file not found

Reply:
>As a workaround we could skip the check for NEON if SSE or AVX is supported.

At configure time or at run-time?

In general, we should support cross-compiling with either gcc or clang.

Reply:
AIUI, the configure check checks to see whether the compiler being used for compiling supports the flag. So any ARM targetted compiler (whether a cross-compiler or a native one) should pass the check.

I think I need another configure check in there for whether that header is available. If the header isn't available, I'll disable neon.

I'll have a crack at a PR for that tomorrow.

Reply:
@amitdo, it's a check in `configure`, so it is a configure time check. :-)

The problem here is that a compilation for Intel architecture uses a compiler which claims to support NEON. I hope that there are no ARM compilers which support AVX. 

@robinwatts, checking for the header is not a save indicator for ARM, because the same header is also available for MIPS and RISCV64. `configure` also has so many checks that it takes a significant time to run. That's why I prefer a solution which only executes the relevant checks: no AVX ... checks for non-Intel, no NEON check for non-ARM.